lib/mix/lib/mix/tasks/test.ex

```diff
@@ -660,6 +703,33 @@ defmodule Mix.Tasks.Test do
     end
   end
 
+  # similar to Mix.Utils.extract_files/2, but returns a list of directly included test files,
+  # that should be not filtered by the test_load_filters, e.g.
+  # mix test test/some_file.exs
+  defp extract_files(paths, pattern) do
+    {files, directly_included} =
+      for path <- paths, reduce: {[], []} do
+        {acc, directly_included} ->
+          case :elixir_utils.read_file_type(path) do
+            {:ok, :directory} ->
+              {[acc, Path.wildcard("#{path}/**/#{pattern}")], directly_included}
+
+            {:ok, :regular} ->
+              {[acc, path], [path | directly_included]}
```

Comment:
```suggestion
              {[Path.wildcard("#{path}/**/#{pattern}") | acc], directly_included}

            {:ok, :regular} ->
              {[path | acc], [path | directly_included]}
```

To avoid generating a weird nested list.